tests/dummy/app/services/session-account.js

```diff
@@ -1,22 +1,14 @@
-import RSVP from 'rsvp';
 import Service, { inject as service } from '@ember/service';
-import { isEmpty } from '@ember/utils';
 
 export default Service.extend({
   session: service('session'),
   store: service(),
 
-  loadCurrentUser() {
-    return new RSVP.Promise((resolve, reject) => {
-      const accountId = this.get('session.data.authenticated.account_id');
-      if (!isEmpty(accountId)) {
-        this.get('store').find('account', accountId).then((account) => {
-          this.set('account', account);
-          resolve();
-        }, reject);
-      } else {
-        resolve();
-      }
-    });
+  async loadCurrentUser() {
+    let accountId = this.get('session.data.authenticated.account_id');
+    if (accountId) {
+      let account = await this.get('store').find('account', accountId);
+      this.account = account;
```

Comment:
should we not keep using `this.set()` here?

Reply:
yes, fixed